subsys/bluetooth/shell/gatt.c

```diff
@@ -1152,8 +1144,8 @@ SHELL_STATIC_SUBCMD_SET_CREATE(gatt_cmds,
 		      cmd_subscribe, 3, 1),
 	SHELL_CMD_ARG(resubscribe, NULL, HELP_ADDR_LE" <CCC handle> "
 		      "<value handle> [ind]", cmd_resubscribe, 5, 1),
-	SHELL_CMD_ARG(write, NULL, "<handle> <offset> <data> [length]",
-		      cmd_write, 4, 1),
+	SHELL_CMD_ARG(write, NULL, "<handle> <offset> <data> [data...]",
```

Comment:
In my opinion it would be better just support <data> as a longer string argument, like adv-data for example.

`bt adv-data 020106 scan-response 03051104`

Reply:
I guess we are also using hex2bin on adv-data, btw I don't think we should interleave mandatory with optional parameters, it should be mandatory follow by optional as the shell will attempt to verify if the minimal parameters given is equal or above mandatory ones but if there are optional in the middle this check no longer works properly.